src/utils/hooks/useAlerts/useAlerts.ts

```diff
@@ -0,0 +1,33 @@
+import { useMemo } from 'react';
+
+import {
+  getAlertsAndRules,
+  silenceFiringAlerts,
+} from '@kubevirt-utils/hooks/useAlerts/utils/utils';
+import useSilences from '@kubevirt-utils/hooks/useSilences/useSilences';
+import {
+  Alert,
+  PrometheusEndpoint,
+  usePrometheusPoll,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { PrometheusRulesResponse } from '@virtualmachines/details/tabs/overview/utils/utils';
+
+type UseAlerts = () => { alerts: Alert[]; loaded: boolean; loadError: unknown };
+
+const useAlerts: UseAlerts = () => {
+  const [response] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint?.RULES,
+  });
+  const { silences } = useSilences();
+
+  const allAlerts = useMemo(() => {
+    const data = (response as PrometheusRulesResponse)?.data;
```

Comment:
why we need as here? what usePrometheusPoll returns?

Reply:
`usePrometheusPoll` returns the `PrometheusResponse` type, whose `data` property has a value of `PrometheusData`. The data property of `PrometheusRulesResponse` has a value of `{ groups?: Group[] }`, which conforms to the shape of the data returned from the `PrometheusEndpoint.RULES` endpoint.